pyrometheus/codegen/python.py

```diff
@@ -466,12 +471,18 @@ def gen_thermochem_code(sol: ct.Solution) -> str:
         falloff_reactions=list(compress(sol.reactions(),
                                         [isinstance(r, ct.FalloffReaction)
                                          for r in sol.reactions()])),
+        falloff_indices=list(compress(range(sol.n_reactions),
+                                      [isinstance(r, ct.FalloffReaction)
+                                       for r in sol.reactions()])),
         non_falloff_reactions=list(compress(sol.reactions(),
                                             [not isinstance(r, ct.FalloffReaction)
                                              for r in sol.reactions()])),
         three_body_reactions=list(compress(sol.reactions(),
                                            [isinstance(r, ct.ThreeBodyReaction)
                                             for r in sol.reactions()])),
+        three_body_indices=list(compress(range(sol.n_reactions),
+                                         [isinstance(r, ct.ThreeBodyReaction)
+                                         for r in sol.reactions()]))
```

Comment:
Rewrite these as list comprehensions of tuples with a conditional:
```
[(i, react) for i, react in enumerate(sol.reactions()) if isinstance(react, ...)]
```
This will avoid the need for separate `indices` arrays.